app/src/main/java/org/dhis2/usescases/teiDashboard/TeiDashboardMobileActivity.java

```diff
@@ -157,57 +181,117 @@ protected void onRestoreInstanceState(Bundle savedInstanceState) {
+                            } else {
+                                binding.filterCounter.setVisibility(View.VISIBLE);
+                                binding.searchFilterGeneral.setVisibility(View.VISIBLE);
+                            }
+                        }
+                    }
+            );
+            //binding.tabLayout.setVisibility(programUid != null ? View.VISIBLE : View.GONE);
             if (fromRelationship)
                 binding.teiPager.setCurrentItem(2, false);
+
+            tabLayoutMediator(binding.teiPager);
         } else {
-            tabletAdapter = new DashboardPagerTabletAdapter(this, getSupportFragmentManager(), programUid, teiUid);
+            tabletAdapter = new DashboardPagerTabletAdapter(this, programUid, teiUid, enrollmentUid);
             currentAdapter = tabletAdapter;
-            binding.teiPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
-                @Override
-                public void onPageScrolled(int i, float v, int i1) {
-                    // nothing
-                }
-
-                @Override
-                public void onPageSelected(int i) {
-                    binding.sectionTitle.setText(tabletAdapter.getPageTitle(i));
-                }
+            binding.teiTablePager.registerOnPageChangeCallback(
+                    new ViewPager2.OnPageChangeCallback() {
+                        @Override
+                        public void onPageSelected(int position) {
+                            switch (position) {
+                                case 1:
+                                    binding.sectionTitle.setText(getString(R.string.dashboard_relationships));
+                                    break;
+                                case 2:
+                                    binding.sectionTitle.setText(getString(R.string.dashboard_notes));
+                                    break;
+                                default:
+                                    binding.sectionTitle.setText(getString(R.string.dashboard_indicators));
+                                    break;
+                            }
+                        }
+                    }
+            );
 
-                @Override
-                public void onPageScrollStateChanged(int i) {
-                    // nothing
-                }
-            });
-            binding.sectionTitle.setText(tabletAdapter.getPageTitle(0));
-            binding.teiPager.setAdapter(tabletAdapter);
-            binding.tabLayout.setVisibility(View.GONE);
+            binding.teiTablePager.setAdapter(tabletAdapter);
             binding.dotsIndicator.setVisibility(programUid != null ? View.VISIBLE : View.GONE);
-            binding.dotsIndicator.setViewPager(binding.teiPager);
+            //binding.dotsIndicator.setViewPager(binding.teiPager); // TODO look into dots Indicator integration with viewPager 2
```

Comment:
???

Reply:
Currently, the library that we are using is locally in the app and has some changes. That version does not support ViewPager2. We need to try to update the library to the new version that supports ViewPager2. However, it not so simple since the library has been completely rewritten in Kotlin.